### PR TITLE
Auto backup

### PR DESCRIPTION
**Pull request by oliexdev**
basic import/export and auto backup function of the database.

### CODE REVIEW

File: android_app/app/src/main/java/com/health/openscale/core/OpenScale.java [hunk 1]
```diff
@@ -385,6 +387,54 @@ public String getFilenameFromUri(Uri uri) {
...
+
+            while ((count = input.read(bytes)) != -1){
+                output.write(bytes, 0, count);
+            }
+        } finally {
+            if (input != null)
+                input.close();
+            if (output != null)
+                output.flush();
+                output.close();
```
Comment (erijo) [inline android_app/app/src/main/java/com/health/openscale/core/OpenScale.java]: Missing {} means this doesn't do what it looks like.

File: android_app/app/src/main/java/com/health/openscale/core/alarm/ReminderBootReceiver.java [hunk 2]
```diff
@@ -27,6 +27,8 @@ public void onReceive(Context context, Intent intent)
     {
         if (intent.hasExtra(AlarmHandler.INTENT_EXTRA_ALARM)) handleAlarm(context);
 
+        if (intent.hasExtra(AlarmBackupHandler.INTENT_EXTRA_BACKUP_ALARM)) handleBackupAlarm(context);
```
Comment (erijo) [inline android_app/app/src/main/java/com/health/openscale/core/alarm/ReminderBootReceiver.java]: I've never gotten the reminder alarm to work. Do you have any idea why that could be?

File: android_app/app/src/main/java/com/health/openscale/core/OpenScale.java (hunk 1 above)
Comment (oliexdev) [reply android_app/app/src/main/java/com/health/openscale/core/OpenScale.java]: true it is fixed.

File: android_app/app/src/main/java/com/health/openscale/core/alarm/ReminderBootReceiver.java (hunk 2 above)
Comment (oliexdev) [reply android_app/app/src/main/java/com/health/openscale/core/alarm/ReminderBootReceiver.java]: hmm not really for me it is working

Comment (oliexdev) [reply android_app/app/src/main/java/com/health/openscale/core/alarm/ReminderBootReceiver.java]: did you disable globally on your smartphone notifications?

**Pull request merged**